Automattic/jetpack
# Tools: Script to rename default branches on GitHub

#### Changes proposed in this Pull Request:

* This script will rename the default branches (or rather the master branches) in a list of repositories.
* For security, by default it will only rename a few non-important repositories.

#### Jetpack product discussion

p9dueE-59T-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Check out this PR locally.
* In the Jetpack repository root, run `tools/gh-rename-branches`.
* This, by default, will only rename three Automattic Jetpack repositories (jetpack-google-analytics-helper, jptest-tools and jetpack-gh-testing), and only after prompting for your GitHub personal access token.
* For more tests, open up the file and un-comment out the test function at the bottom of the script after commenting out the main function. This will return the name of the default branch in each of the given repositories.

### Questions / assumptions:

* This script assumes you've created an access token via Github and added it when prompted. 
* This script only renames the mirror repos , assuming we want to rename Jetpack on it's own using the GitHub UI - but it could be added here too.
* Currently the script uses CURL as it may be more widely used than the GitHub CLI, but both options exist. One benefit (or drawback) of using the GH CLI is that it actually renames the branches one by one - entering `q` allows it to continue to the next branch. So it's technically a more manual process, but that also means the script execution can be exited easily at any stage.
* One thing this script doesn't have is any error-catching tests, so that is something to look at next.

The intention of this PR is to get input on the script itself - it doesn't need to be merged (not at this stage anyway).
